### PR TITLE
Bringing back compatibility with python 3.4

### PR DESCRIPTION
**Pull request by rtobar**
The latest release of zeroconf in PyPI (0.21.0) breaks compatibility with python 3.4 due to an unstated dependency on the ``typing`` package. This patch adds that dependency and issues a new minor version of zeroconf.

### CODE REVIEW

File: zeroconf.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 __author__ = 'Paul Scott-Murphy, William McBrine'
 __maintainer__ = 'Jakub Stasiak <jakub@stasiak.at>'
-__version__ = '0.21.0'
+__version__ = '0.21.1'
```
Comment (jstasiak) [inline zeroconf.py]: This is out of scope, let us handle that externally.

File: setup.py [hunk 2]
```diff
@@ -14,6 +15,12 @@
     .strip().strip('\'"')
 )
 
+install_requires = [
+    'ifaddr'
+]
+if sys.version_info[:2] < (3, 5):
+    install_requires.append('typing')
```
Comment (jstasiak) [inline setup.py]: This makes it so that binary packages created on Python 3.5+ won't have the typing dependency which is not what we want. 
https://github.com/eventlet/eventlet/commit/b4b20c459af34d54b52e98a72fe68f6e5143398f shows the way to do it.

File: zeroconf.py (hunk 1 above)
Comment (rtobar) [reply zeroconf.py]: Sure, I'll take that out.

File: setup.py (hunk 2 above)
Comment (rtobar) [reply setup.py]: Isn't the `typing` module included in 3.5+? And if I understand correctly, binary wheels are specific to python versions, so you would be producing them independently for each of them, and therefore the dependency should appear correctly.

The difference then would only on how the dependency is stated and processed (I think I'm still stuck to the old ways of setuptools); I'll change it now.

Comment (jstasiak) [reply setup.py]: Yes, it's included in Python 3.5. The correct requirement scoping will only install the extra dependency on Python 3.4.

We provide universal wheels (see https://github.com/jstasiak/python-zeroconf/blob/b03cee348973469e9ebfce6e9b0e0a367c146401/setup.cfg#L2)

Comment (rtobar) [reply setup.py]: Ohh, I didn't realize that, sorry, something new I learned today...

**Comment by rtobar** [comment]
Sorry for the unnecessary extra noise!

**Comment by jstasiak** [comment]
No worries, thanks for the patch!

**Comment by jstasiak** [comment]
For the record this patch fixes https://github.com/jstasiak/python-zeroconf/issues/139.
